### PR TITLE
Generalize Reference Data Exclusion Logic

### PR DESCRIPTION
**Pull request by AFg6K7h4fhy2**
For the scope of this PR, please refer to issue #143 .

### CODE REVIEW

**Comment by AFg6K7h4fhy2** [comment]
It's amazing how the human brain can gloss over and ignore reality.

I spent about 40 minutes looks for a reason why `RcppTOML::parseTOML(exclude_data_path_toml)` was producing NULL values for the name `ref_date` only to find that there was simply inconsistency with the date in the `toml` file and the one I entered. I must have looked at the `toml` file for at least a minute and even when I was "looking for errors", _the obvious_ did not percolate to the realm of my conscious mind until just a moment ago.

File: src/code/gen_forecast_data.R [hunk 1]
```diff
@@ -54,25 +54,29 @@ hub_content <- hubData::connect_hub(base_hub_path)
 # check if the reference date is the first
 # week for the season (2024-11-23), if so
 # exclude some locations
```
Comment (sbidari) [inline src/code/gen_forecast_data.R:56]:
```suggestion
# check if the reference date has any
# exclusions and exclude specified locations (if any)
```

File: src/code/gen_map_data.R [hunk 2]
```diff
@@ -95,21 +95,21 @@ if (length(missing_pop_columns) > 0) {
 # check if the reference date is the first
 # week for the season (2024-11-23), if so
 # exclude some locations
```
Comment (sbidari) [inline src/code/gen_map_data.R:97]:
```suggestion
# check if the reference date has any
# exclusions and exclude specified locations if any
```

**Review by sbidari — COMMENTED** [review]
Looks good to me, just a minor change in both files.

**Review by sbidari — CHANGES_REQUESTED** [review]
You can also remove the now duplicated json file `2024-11-23-exclude-locations.json`

**Review by sbidari — APPROVED** [review]

**Pull request merged**